akka-contrib/src/main/scala/akka/contrib/pattern/DistributedPubSubMediator.scala

```diff
@@ -548,6 +574,10 @@ class DistributedPubSubMediator(
       content = bucket.content + (key -> ValueHolder(v, valueOption))))
   }
 
+  def getCurrentTopics(): Map[String, ActorRef] = {
+    context.children.map(ref ⇒ (URLDecoder.decode(ref.path.name, "utf-8"), ref)).toMap
+  }
```

Comment:
I think I have misguided you. Sorry. The child actors are only the topics that have been registered at a specific node. We want all replicated topics. We must grab them from the registry. Something like:

  def getCurrentTopics(): Set[String] = {
    val topicPrefix = self.path.toStringWithoutAddress
    (for {
      (_, bucket) ← registry
      (key, value) ← bucket.content
      if key.startsWith(topicPrefix)
      topic = key.substring(topicPrefix.length + 1)
      if !topic.contains('/') // exclude group topics
    } yield URLDecoder.decode(topic, "utf-8")).toSet
  }